toml/_parser.ts

```diff
@@ -229,25 +233,25 @@ export function deepAssign(
 // ---------------------------------
 
 // deno-lint-ignore no-explicit-any
-function or<T extends readonly ParserComponent<any>[]>(
+function or<T extends readonly ParserComponent<TOMLValue>[]>(
   parsers: T,
 ): ParserComponent<
   ReturnType<T[number]> extends ParseResult<infer R> ? R : Failure
 > {
-  return (scanner: Scanner) => {
+  return ((scanner: Scanner) => {
     for (const parse of parsers) {
       const result = parse(scanner);
       if (result.ok) return result;
     }
     return failure();
-  };
+  }) as unknown as any;
```

Comment:
This needs `// deno-lint-ignore no-explicit-any` directive before this line to pass lint check. See the `deno lint` output